android/services/mongodb-remote-perftests/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/performance/SyncR2LOnlyPerformanceTestDefinitions.kt

```diff
@@ -0,0 +1,269 @@
+                    val numSyncedIds = Tasks.await(sync.syncedIds).size
+                    val numLocalDocs = Tasks.await(sync.count())
+
+                    SyncPerformanceTestUtils.assertIntsAreEqualOrThrow(
+                        numSyncedIds, numDocs, "Number of Synced Ids")
+                    SyncPerformanceTestUtils.assertIntsAreEqualOrThrow(
+                        numLocalDocs.toInt(), numDocs, "Number of Local Documents")
+
+                    // Disconnect the DataSynchronizer and wait
+                    // for the underlying streams to close
+                    ctx.testNetworkMonitor.connectedState = false
+                    while (ctx.testDataSynchronizer.areAllStreamsOpen()) {
+                        testHarness.logMessage("waiting for streams to close")
+                        Thread.sleep(1000)
+                    }
+                },
+                testDefinition = { ctx, _, _ ->
+                    // Reconnect the DataSynchronizer, and wait for the streams to reopen. The
+                    // stream being open indicates that the doc configs are now set as stale.
+                    // Check every 10ms so we're not doing too much work on this thread, and
+                    // don't log anything, so as not to pollute the test results with logging
+                    // overhead.
+                    ctx.testNetworkMonitor.connectedState = true
+                    var counter = 0
+                    while (!ctx.testDataSynchronizer.areAllStreamsOpen()) {
+                        Thread.sleep(10)
+
+                        // if this hangs longer than 30 seconds, throw an error
+                        counter += 1
+                        if (counter > 3000) {
+                            testHarness.logMessage("stream never opened after reconnect")
+                            error("stream never opened after reconnect")
+                        }
+                    }
+
+                    // Do the sync pass that will perform the stale document fetch
+                    val syncPassSucceeded = ctx.testDataSynchronizer.doSyncPass()
+
+                    // Perform syncPass() and halt the test if the pass fails
+                    if (!syncPassSucceeded) {
+                        error("sync pass failed")
+                    }
+                },
+                afterEach = { ctx, numDocs, _ ->
+                    // Verify that the test did indeed synchronize the updates locally
+                    SyncPerformanceTestUtils.assertLocalAndRemoteDBCount(ctx, numDocs)
+                }
+            )
+        }
+
+        fun testSyncPass(testHarness: SyncPerformanceIntTestsHarness, runId: ObjectId) {
+            // Run doTestSyncPass() for all changeEvent Percentages found in SyncPerfTestUtils
+            SyncPerformanceTestUtils.getChangeEventPercentages().forEach {
+                doTestSyncPass(testHarness, runId, it)
+            }
+        }
+
+        private fun doTestSyncPass(
```

Comment:
[task] per the performance testing spec, percentage of conflicts also needs to be a parameter here (I slack DM'ed the spec link). the purpose of this is to measure the performance of the conflict-handling codepaths

e.g. if you're supposed to test where 1% of the documents are conflicted, in the setup you'll need to locally update 1% of the documents you updated remotely, and define some simple conflict handler.